sql/hive-thriftserver/v1.2.1/src/main/java/org/apache/hive/service/cli/ColumnValue.java

```diff
@@ -195,7 +195,7 @@ public static TColumnValue toTColumnValue(Type type, Object value) {
     case INTERVAL_DAY_TIME_TYPE:
       return stringValue((HiveIntervalDayTime) value);
     case DECIMAL_TYPE:
-      return stringValue(((HiveDecimal)value));
+      return stringValue(((BigDecimal)value).toPlainString());
     case BINARY_TYPE:
       String strVal = value == null ? null : UTF8String.fromBytes((byte[])value).toString();
       return stringValue(strVal);
```

Comment:
Please remove these lines because we do not need it:
https://github.com/apache/spark/pull/23899/files#diff-15f177f5340d8433fafc231afa298b53R26
https://github.com/apache/spark/pull/23899/files#diff-15f177f5340d8433fafc231afa298b53R143-R149

Reply:
done

Reply:
done